controllers/certificaterequest/issue_certificate.go

```diff
@@ -210,3 +214,32 @@ func (r *CertificateRequestReconciler) IssueCertificate(reqLogger logr.Logger, c
 
 	return nil
 }
+
+func (r *CertificateRequestReconciler) FindZoneIDForChallenge(namespace string) (string, error) {
+	dnsZones := hivev1.DNSZoneList{}
+	err := r.Client.List(context.TODO(), &dnsZones, &client.ListOptions{Namespace: namespace})
+	if err != nil {
+		return "", err
+	}
+
+	if len(dnsZones.Items) != 1 {
+		return "", fmt.Errorf("%d dnsZone objects in a specific namespace found, expected 1 dnsZone", len(dnsZones.Items))
+	}
+
+	dnsZoneStatus := dnsZones.Items[0].Status
+	if dnsZoneStatus.AWS != nil {
+		if dnsZoneStatus.AWS.ZoneID != nil {
+			return filepath.Base(*dnsZoneStatus.AWS.ZoneID), nil //the format of that ID is (/hostedzone/<something>) so, we don't care about >/. Here, we only want to grab the stuff after the last
```

Comment:
Nit: can we make this comment a bit more informative? Something along the lines of the following might make it clearer:
```suggestion
			// The format of this field is "/hostedzone/<hostedZoneID>"
			// Since we only want to return the hostedZoneID, we can use the filepath utils to remove the prefix before returning
			return filepath.Base(*dnsZoneStatus.AWS.ZoneID), nil 
```